TestApp generation should add necessary imports

Function TestApp does make use of `os`, `ioutil`, `noarch` and `testing` packages. We should make sure that those imports are always included.

```
import "io/ioutil"
import "os"
import "testing"
import "github.com/elliotchance/c2go/noarch"

func TestApp(t *testing.T) {
   os.Chdir("../../..")
   ioutil.WriteFile("build/stdin", []byte{'7'}, 0777)
   stdin, _ := os.Open("build/stdin")
   noarch.Stdin = noarch.NewFile(stdin)
   main()
}
```

It is possible to create trivial programs where the `main` function do not refer to those imports.
```
int main() {
    return 97;
}
```
=> `./main.go:21:2: undefined: noarch`


Low priority since any non-trivial C program will import "os" and "noarch".

--- a/main.go
+++ b/main.go
@@ -217,7 +217,7 @@ func Start(args ProgramArgs) (err error) {
 
 	p := program.NewProgram()
 	p.Verbose = args.verbose
-	p.OutputAsTest = true // args.outputAsTest
+	p.OutputAsTest = args.outputAsTest
 
 	// Converting to nodes
 	if args.verbose {

--- a/transpiler/declarations.go
+++ b/transpiler/declarations.go
@@ -251,7 +251,7 @@ func transpileVarDecl(p *program.Program, n *ast.VarDecl) (
 		switch name {
 		// Below are for macOS.
 		case "__stdinp", "__stdoutp":
-			p.AddImports("github.com/elliotchance/c2go/noarch", "os")
+			p.AddImport("github.com/elliotchance/c2go/noarch")
 			p.AppendStartupExpr(
 				util.NewBinaryExpr(
 					goast.NewIdent(name),
@@ -265,7 +265,7 @@ func transpileVarDecl(p *program.Program, n *ast.VarDecl) (
 			// Below are for linux.
 		case "stdout", "stdin", "stderr":
 			theType = "*noarch.File"
-			p.AddImports("github.com/elliotchance/c2go/noarch", "os")
+			p.AddImport("github.com/elliotchance/c2go/noarch")
 			p.AppendStartupExpr(
 				util.NewBinaryExpr(
 					goast.NewIdent(name),

--- a/transpiler/transpiler.go
+++ b/transpiler/transpiler.go
@@ -33,6 +33,7 @@ func TranspileAST(fileName, packageName string, p *program.Program, root ast.Nod
 	if p.OutputAsTest {
 		p.AddImport("testing")
 		p.AddImport("io/ioutil")
+		p.AddImport("os")
 
 		// TODO: There should be a cleaner way to add a function to the program.
 		// This code was taken from the end of transpileFunctionDecl.

